### DISCUSSION TITLE
File naming [Cast List] exceedingly long filenames

### DISCUSSION THREAD

**Discussion by jfreiman** [CodyBerenson/PGMA-Modernized#324] [General]
I only recently noticed that many of my feature videos aren't being backed up my Synology Drive (Windows 11) because the file length is goes beyond the 247 character limit.

I know the [cast] filename length was brought up before, but I'm curious about one thing in particular.

If I am just using the [cast] option to fix/standardize on using a model's specific name instead of the one used by, for example, GEVI, can I just add ONLY the models names I wish to replace with the alternate name and leave the other models out of the filename? Will they still be added?

If I only need to add a few names to [cast] instead of the entire cast; I may be able to resolve the Synology backup issue for most files.

This of course won't work for every title, but it's a start.

** as a matter of fact, while the file(s) is saved in the proper folder(s), at least one of them isn't even seen by Plex - likely because of the file name length. 🤔

**Comment by JPH71** [comment]
Give it a try and if it doesn't work, I will amend the code..

**Reply by jfreiman** [reply]
> Give it a try and if it doesn't work, I will amend the code..
> […](#)
> On Wed, 27 Nov 2024, 03:31 John Freiman, ***@***.***> wrote: I only recently noticed that many of my feature videos aren't being backed up my Synology Drive (Windows 11) because the file length is goes beyond the 247 character limit. I know the [cast] filename length was brought up before, but I'm curious about one thing in particular. If I am just using the [cast] option to fix/standardize on using a model's specific name instead of the one used by, for example, GEVI, can I just add ONLY the models names I wish to replace with the alternate name and leave the other models out of the filename? Will they still be added? If I only need to add a few names to [cast] instead of the entire cast; I may be able to resolve the Synology backup issue for most files. This of course won't work for every title, but it's a start. ** as a matter of fact, while the file(s) is saved in the proper folder(s), at least one of them isn't even seen by Plex - likely because of the file name length. 🤔 — Reply to this email directly, view it on GitHub <#324>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AKI3AKOTGDM326EJXINZWGD2CUVHNAVCNFSM6AAAAABSRY32LGVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGU3DCOBTGQ> . You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

I want to test this out as well, but I can't at the moment because I can't get PGMA to work any longer. :(
https://github.com/CodyBerenson/PGMA-Modernized/discussions/323#discussion-7561745

**Comment by ms52538** [comment]
John, great question(s) - this has always been a murky area for to fully understand. How are you determining what name to use as you standardize on an actor, who may have several aliases in the International Adult Film Database (which is straight forward with them choosing a primary name and then listing aliases) as compared to Gevi or a blog site that uses a specific alias.  I gather you have employed a methodology you are attempting to use. For me, my attempts to list actors names in the format to identify and get headshots, seem to fail in getting matches made by the IAFD agent, but if I remove them, the media then can make the match.

**Reply by jfreiman** [reply]
> John, great question(s) - this has always been a murky area for to fully understand. How are you determining what name to use as you standardize on an actor, who may have several aliases in the International Adult Film Database (which is straight forward with them choosing a primary name and then listing aliases) as compared to Gevi or a blog site that uses a specific alias. I gather you have employed a methodology you are attempting to use.

I use whatever is the primary name listed in IAFD.  IAFD is also very responsive and helpful in adding Alias' or making other scene, feature or other cast updates - I always send links or image captures to back up my requests. ;)

**Reply by jfreiman** [reply]
> I gather you have employed a methodology you are attempting to use. For me, my attempts to list actors names in the format to identify and get headshots, seem to fail in getting matches made by the IAFD agent, but if I remove them, the media then can make the match.

Yes, my methodology is 2 fold.

I really do want photos of the Cast and ideally, the IAFD mentioning of Top/Bottom, Alias used for the particular video etc.

But I also want the models to be easily found.... retried, when you click on their cast image in a video (within the collection).

I used to take a lot of pride in all the successful matches (thanks to the hard work of everyone who participates in developing PGMA!) in how complete not only the cast images were, but also that GEVI's model names mess (they seem to always use the Alias for the video/studio, not even the GEVI 'primary' name) -  

Regarding blog sites, if they use a different model name, and you can verify that it's the same model as in IAFD they will (or do for me) update the Alias' list so that your model, after the next refresh, will populate with the data and image. IAFD will use the link to the video in the blog and update their info - I will say that I give them very detailed info so they know they are making good and accurate changes.

I can not tell you how accommodating IAFD has been to me and my updates, and corrections. They are part of the reason I have such an easy to search Plex database.

**IAFD and other PGMA agents  have been broken on my Plex install since about 2 weeks after the last PGMA fix/update. (2 and a half months).

With over 17K adult features and 15k adult scenes and growing, I pay a lot of attention to detail and have a process that prevents me from introducing duplicates, etc. --- That said, I avoid a lot of headaches by not introducing any (or very few) fan site videos.

Modology
Find video on IAFD. If its not there, I look go to the model's IAFD page and see if it has a different title, studio, etc. I it's there I alter the video's name, date etc.  In the case of it being released by another studio then the one watermarked on the videos, I use that Studio's name in the mp4 metadata to overwrite what PGMA scrapes. ( @Copyright )
If I can't find it on IAFD, I look for it in GEVI and enter the GEVI # and if the names match, I'm done. If they don't, I verify that the IAFD name is in fact an Alias on GEVI.
If it is, great! If not, I write to IAFD and see if they can add an alias or consider updating the models primary name - In the times I've done this I've also been able to use the models 'twitter'/social account as 'proof' as well as a link to thee Studios video page.

I do not have a way of contacting GEVI - although I think messaged them before, they said submitted edits were possible, but they stopped short of providing me with a link or mechanism to request changes. :(
I then go to GEVI and
